File: python/fedml/core/distributed/communication/base_com_manager.py
from abc import abstractmethod

from .message import Message
from .observer import Observer


class BaseCommunicationManager(object):
    @abstractmethod
    def send_message(self, msg: Message):
        pass

    @abstractmethod
    def add_observer(self, observer: Observer):
        pass

    @abstractmethod
    def remove_observer(self, observer: Observer):
        pass

    @abstractmethod
    def handle_receive_message(self):
        pass

    @abstractmethod
    def stop_receive_message(self):
        pass

File: javascript/src/fedml/core/distributed/communication/base_com_manager.ts
import type { Message } from './message';
import type { Observer } from './observer';

export abstract class BaseCommunicationManager {
  abstract send_message(msg: Message): any;

  abstract add_observer(observer: Observer): any;

  abstract remove_observer(observer: Observer): any;

  abstract handle_receive_message(): any;

  abstract stop_receive_message(): any;
}
